app/controllers/manufacturers_controller.rb

```diff
@@ -32,5 +46,13 @@ def destroy
     def manufacturer_params
       params.require(:manufacturer).permit(:username, :password, :name, :email, :telephone)
     end
+
+    def manufacturer_update_params
+      params.require(:manufacturer).permit(:email, :telephone)
+    end
+
+    def manufacturer_update_params
```

Comment:
Why is the function defined two times?

Reply:
because when updating we only want to update a users email or telephone.